tfe/provider_test.go

```diff
@@ -125,6 +125,155 @@ func TestProvider_versionConstraints(t *testing.T) {
 	}
 }
 
+func TestProvider_locateConfigFile(t *testing.T) {
+	originalHome := os.Getenv("HOME")
+	originalTfCliConfigFile := os.Getenv("TF_CLI_CONFIG_FILE")
+	originalTerraformConfig := os.Getenv("TERRAFORM_CONFIG")
+	reset := func() {
+		os.Setenv("HOME", originalHome)
+		if originalTfCliConfigFile != "" {
+			os.Setenv("TF_CLI_CONFIG_FILE", originalTfCliConfigFile)
+		} else {
+			os.Unsetenv("TF_CLI_CONFIG_FILE")
+		}
+		if originalTerraformConfig != "" {
+			os.Setenv("TERRAFORM_CONFIG", originalTerraformConfig)
+		} else {
+			os.Unsetenv("TERRAFORM_CONFIG")
+		}
+	}
+	defer reset()
+
+	// Use a predictable value for $HOME
+	os.Setenv("HOME", "/Users/someone")
+
+	setup := func(tfCliConfigFile, terraformConfig string) {
+		os.Setenv("TF_CLI_CONFIG_FILE", tfCliConfigFile)
+		os.Setenv("TERRAFORM_CONFIG", terraformConfig)
+	}
+
+	cases := map[string]struct {
+		tfCliConfigFile string
+		terraformConfig string
+		result          string
+	}{
+		"has TF_CLI_CONFIG_FILE": {
+			tfCliConfigFile: "~/.terraform_alternate/terraformrc",
+			terraformConfig: "",
+			result:          "~/.terraform_alternate/terraformrc",
+		},
+		"has TERRAFORM_CONFIG": {
+			tfCliConfigFile: "",
+			terraformConfig: "~/.terraform_alternate_rc",
+			result:          "~/.terraform_alternate_rc",
+		},
+		"has both env vars": {
+			tfCliConfigFile: "~/.from_TF_CLI",
+			terraformConfig: "~/.from_TERRAFORM_CONFIG",
+			result:          "~/.from_TF_CLI",
+		},
+		"has neither env var": {
+			tfCliConfigFile: "",
+			terraformConfig: "",
+			result:          "/Users/someone/.terraformrc", // expect tests run on unix
+		},
+	}
+
+	for name, tc := range cases {
+		setup(tc.tfCliConfigFile, tc.terraformConfig)
+
+		fileResult := locateConfigFile()
+		if tc.result != fileResult {
+			t.Fatalf("%s: expected config file at %s, got %s", name, tc.result, fileResult)
+		}
+	}
+}
+
+func TestProvider_cliConfig(t *testing.T) {
+	// This only tests the behavior of merging various combinations of
+	// (credentials file, .terraformrc file, absent). Locating the .terraformrc
+	// file is tested separately.
+	originalHome := os.Getenv("HOME")
+	originalTfCliConfigFile := os.Getenv("TF_CLI_CONFIG_FILE")
+	reset := func() {
+		os.Setenv("HOME", originalHome)
+		if originalTfCliConfigFile != "" {
+			os.Setenv("TF_CLI_CONFIG_FILE", originalTfCliConfigFile)
+		} else {
+			os.Unsetenv("TF_CLI_CONFIG_FILE")
+		}
+	}
+	defer reset()
+
+	// Summary of fixtures: the credentials file and .terraformrc file each have
+	// credentials for two hosts, and they both have credentials for
+	// app.terraform.io
+	hasCredentials := "test-fixtures/env/home"
+	noCredentials := "test-fixtures/env/empty"
+	terraformrc := "test-fixtures/env/terraformrc"
+	noTerraformrc := "test-fixtures/env/nothing"
```

Comment:
I know i'm the one who made this directory path, so I blame my older self 😆 . I think if I was looking at the directory path, i'd be confused what "env" referred to. Could we change these paths to be `test-fixtures/tf-env-files/*`

Reply:
And nit comment, could we be more specific for the `empty` and `nothing`? So like `test-fixtures/tf-env-files/no-terraformrc` and `test-fixtures/tf-env-files/no-credentials`

Reply:
👍🏼 !

I went with `test-fixtures/cli-config-files/*` for the path name.